atomix/cluster/src/test/java/io/atomix/raft/DeterministicSingleThreadContext.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright © 2020 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.raft;
+
+import static org.junit.Assert.fail;
+
+import io.atomix.utils.concurrent.Scheduled;
+import io.atomix.utils.concurrent.ScheduledFutureImpl;
+import io.atomix.utils.concurrent.ThreadContext;
+import java.time.Duration;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import org.jmock.lib.concurrent.DeterministicScheduler;
+
+public class DeterministicSingleThreadContext implements ThreadContext {
+
+  private final DeterministicScheduler deterministicScheduler;
+  private final Consumer<Throwable> uncaughtExceptionObserver;
+
+  public DeterministicSingleThreadContext(
+      final DeterministicScheduler executor, final Consumer<Throwable> uncaughtExceptionObserver) {
+    deterministicScheduler = executor;
+    this.uncaughtExceptionObserver = uncaughtExceptionObserver;
+  }
+
+  public DeterministicScheduler getDeterministicScheduler() {
+    return deterministicScheduler;
+  }
+
+  public static ThreadContext createContext(
```

Comment:
Normally if you have a static factory method I would expect you make the ctor private.